domain/operation/service/start.go

```diff
@@ -88,30 +93,57 @@ func (s *Service) AddExecOperation(
 		leaderUnits = append(leaderUnits, leaderUnitName)
 	}
```

Comment:
This function is too large to test and grok in one sitting. Move this logic out.